data/templates/frr/isis.frr.tmpl

```diff
@@ -116,18 +116,33 @@ router isis VyOS {{ 'vrf ' + vrf if vrf is defined and vrf is not none }}
 {%     endfor %}
 {%   endfor %}
 {% endif %}
-{% if redistribute is defined and redistribute.ipv4 is defined and redistribute.ipv4 is not none %}
-{%   for protocol in redistribute.ipv4 %}
-{%     for level, level_config in redistribute.ipv4[protocol].items() %}
-{%       if level_config.metric is defined and level_config.metric is not none %}
+{% if redistribute is defined %} 
+{%   if redistribute.ipv4 is defined and redistribute.ipv4 is not none %}
+{%     for protocol in redistribute.ipv4 %}
```

Comment:
Please use `{%     for protocol, protocol_options in redistribute.ipv4.items() %}` which makes the remaining logic easier to read and maintain.

Reply:
I am not *quite* understanding, but let me look at it a few more times to see what you mean. I'm sure your way is better, just trying to wrap my mind around it :)

Reply:
Made the change sir. It works.

Thank you :)